src/ui/utils/sources.ts

```diff
@@ -1,118 +1,63 @@
 import { EntityId } from "@reduxjs/toolkit";
-import { newSource, SourceKind } from "@replayio/protocol";
+import { newSource } from "@replayio/protocol";
+import { ArrayMap, assert } from "protocol/utils";
 import { SourceDetails } from "ui/reducers/sources";
-import newGraph from "./graph";
-
-export const keyForSource = (source: newSource): string => {
-  return `${source.url!}:${source.contentHash}`;
-};
 
 export const newSourcesToCompleteSourceDetails = (
   newSources: newSource[]
 ): Record<EntityId, SourceDetails> => {
-  const prettyPrinted = newGraph();
-  const canonical = newGraph();
-  const corresponding: Record<string, string[]> = {};
   const newSourcesById: Record<string, newSource> = {};
-  const byKind = {} as Record<SourceKind, newSource[]>;
-
-  // Canonical links can go across multiple links
-  const findCanonicalId = (id: string) => {
-    let current = id;
-    let nextNode = canonical.from(current)?.[0];
-
-    while (nextNode && nextNode !== current) {
-      current = nextNode;
-      nextNode = canonical.from(current)?.[0];
-    }
-
-    return current;
-  };
-
-  const generated = newGraph();
-
-  // We'll iterate through the whole list once, and perform three setup operations per source:
```

Comment:
This code is incredibly confusing, even for me, and I wrote it. I think it's fine if we rework it to be more efficient, but not at the cost of removing all explanations of how it works. I would like to understand:

1) Are these changes more performant? How much more?
2) What did we change about how we process sources?
3) Can we leave some explanation about the trickier parts of this code? It can live in the commit, or in separate documentation, or inline, but I think future us will need it somewhere, at some point.

Reply:
1. I didn't rewrite this to be more performant but to be less confusing. I think it's rather straightforward now.
2. no more `canonicalId` and the links between pretty-printed and minified sources are not removed from `generated`/`generatedFrom` anymore
3. I've added a few comments and could add more if you tell me which parts you find confusing